src/main/java/org/commcare/formplayer/session/FormSession.java

```diff
@@ -183,6 +190,30 @@ public FormSession(UserSqlSandbox sandbox,
         }
     }
 
+    public void updateFormTitle(String locale) {
+        session.setTitle(getLocalizedFormTitle(locale));
+    }
+
+    private String getLocalizedFormTitle(String locale) {
+        if (locale == null
+                || !Localization.getGlobalLocalizerAdvanced().hasLocale(locale)
+                || Localization.getCurrentLocale().equals(locale)) {
+            return Localization.getWithDefault(getFormTitleLocaleKey(), formDef.getTitle());
+        } else {
+            return StringUtils.defaultIfBlank(
+                    Localization.getGlobalLocalizerAdvanced().getText(getFormTitleLocaleKey(), locale),
+                    formDef.getTitle());
+        }
+    }
+
+    private String getFormTitleLocaleKey() {
+        if (commandId != null && !commandId.equals("")) {
+            String preparedCommandId = commandId.replace("-", "");
+            return "forms." + preparedCommandId;
+        }
+        return "";
+    }
```

Comment:
there is no guarantee  in Xform spec that forms locale key will follow this format (it can be `dianasour.m0f0` for exmaple). This should be using the same logic as we use to load the display text for menus instead.

Reply:
Noted